src/lib/core/immer-adapter/common/immutable-state-context.ts

```diff
@@ -39,4 +46,42 @@ export class ImmutableStateContext<T extends object> implements StateContext<T>
   public dispatch(actions: any | any[]): Observable<void> {
     return this.ctx.dispatch(actions);
   }
+
+  private clone<T = any>(obj: T): T {
+    let copy: any;
```

Comment:
Move this declaration under the first `if` condition, after `obj == null`, please